sphinx_gallery/gen_rst.py

```diff
@@ -284,34 +287,21 @@ def get_md5sum(src_file):
     return src_md5
 
 
-def check_md5sum_change(src_file):
-    """Returns True if src_file has a different md5sum"""
+def md5sum_is_current(src_file):
+    """Returns True if src_file has the on disk stored md5sum"""
 
     src_md5 = get_md5sum(src_file)
 
     src_md5_file = src_file + '.md5'
-    src_file_changed = True
     if os.path.exists(src_md5_file):
         with open(src_md5_file, 'r') as file_checksum:
             ref_md5 = file_checksum.read()
         if src_md5 == ref_md5:
-            src_file_changed = False
-
-    if src_file_changed:
-        with open(src_md5_file, 'w') as file_checksum:
-            file_checksum.write(src_md5)
-
-    return src_file_changed
+            return True
+        else:
+            os.remove(src_md5_file)
```

Comment:
I can't think of the counter example now. But my idea was that if the example failed one shall not find the md5 file on disk to trigger rebuilds. But this might be an old part of the check code that is not needed anymore.